From 65b3e7ba64297bdd49c83ff621d96c4e9e638134 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Thu, 21 Oct 2021 11:00:28 +0000
Subject: [PATCH] GCC: fix error with base::Environment name retrieval in thread restrictions.

Fix build error:
    ../../base/threading/thread_restrictions.h:483:35: error: ‘Environment’ has not been declared
      483 |   friend bool ::HasWaylandDisplay(Environment* env);  // crbug.com/1246928
          |                                   ^~~~~~~~~~~
     ../../base/threading/thread_restrictions.h:483:51: error: ‘bool HasWaylandDisplay(int*)’ should have been declared inside ‘::’
      483 |   friend bool ::HasWaylandDisplay(Environment* env);  // crbug.com/1246928
          |                                                   ^

Bug: 819294
Change-Id: Ibec6f1290e78f022e062e51fc66ac58222add772
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3231819
Reviewed-by: Wez <wez@chromium.org>
Commit-Queue: José Dapena Paz <jdapena@igalia.com>
Cr-Commit-Position: refs/heads/main@{#933814}
---

diff --git a/base/threading/thread_restrictions.h b/base/threading/thread_restrictions.h
index 0f02929..aca174d 100644
--- a/base/threading/thread_restrictions.h
+++ b/base/threading/thread_restrictions.h
@@ -480,7 +480,7 @@
   friend bool PathProviderWin(int, FilePath*);
   friend bool chromeos::system::IsCoreSchedulingAvailable();
   friend int chromeos::system::NumberOfPhysicalCores();
-  friend bool ::HasWaylandDisplay(Environment* env);  // crbug.com/1246928
+  friend bool ::HasWaylandDisplay(base::Environment* env);  // crbug.com/1246928
 
   ScopedAllowBlocking(const Location& from_here = Location::Current());
   ~ScopedAllowBlocking();
